### DISCUSSION TITLE
Twig header function

### DISCUSSION THREAD

**Discussion by I-Valchev** [bolt/core#1332] [Ideas / Feature requests]
Have a `header` function, either within Bolt or as an extension, that will set an HTTP header.

Example use cases incldue:
custom cache header
different response type (e.g. for serving json properly)

If it's an extension, the response type can be used to show the page in json at its own slug, for example:
/page/1 -> html (as is)
/page/1.json -> json

With a bit of imagination, this could also be used to create custom API endpoints in Twig, rather than having to extend and work with api-platform itself.

What do you think @bobdenotter ?

**Comment by bobdenotter** [comment]
I think you can already do this: Make a Widget extension with target 'nowhere', and in it's `invoke` you can modify the response headers. For example: 

https://github.com/bolt/core/blob/master/src/Widget/BoltHeaderWidget.php
